types/datasource.d.ts

```diff
@@ -76,28 +90,238 @@ import {IsolationLevel, Transaction} from './transaction-mixin';
  */
 export declare class DataSource extends EventEmitter {
   name: string;
-  settings: Options;
+  settings: ConnectorSettings;
 
   initialized?: boolean;
   connected?: boolean;
   connecting?: boolean;
+  
+  /**
+   * {@inheritDoc DataSource.connector}
+   * @deprecated Use {@link DataSource.connector} instead.
+   */
+  adapter?: Connector;
 
+  /**
+   * Connector instance.
+   */
   connector?: Connector;
 
-  definitions: {[modelName: string]: ModelDefinition};
+  modelBuilder: ModelBuilder;
 
+  models: Record<string, typeof ModelBase>;
+  
+  definitions: {[modelName: string]: ModelDefinition};
+  
   DataAccessObject: AnyObject & {prototype: AnyObject};
+  
+  pendingConnectCallbacks?: Callback[];
+
+  /**
+   * Log benchmarked message.
+   * 
+   * @remarks
+   * This class method is defined as the attached connector's
+   * {@link Connector.log | log} class method.
+   * 
+   * @param sql 
+   * @param t Start time 
+   */
+  private log(sql: string, t: number): void;
 
-  constructor(name: string, settings?: Options, modelBuilder?: ModelBuilder);
+  private _queuedInvocations: number;
 
-  constructor(settings?: Options, modelBuilder?: ModelBuilder);
+  private _operations: Record<string, OperationOptions>;
+
+  operations(): Record<string, OperationOptions>;
+
+  defineOperation(name: string, options: OperationOptions, fn: Function): void;
```

Comment:
These look like they're meant exclusively for "HTTP Remoting" operations as well. We may want to mark them as deprecated.